src/PropertyProcessor/Property/ConstProcessor.php

```diff
@@ -34,13 +34,17 @@ public function process(string $propertyName, JsonSchema $propertySchema): Prope
             $json['description'] ?? '',
         );
 
-        return $property
-            ->setRequired(true)
-            ->addValidator(new PropertyValidator(
-                $property,
-                '$value !== ' . var_export($json['const'], true),
-                InvalidConstException::class,
-                [$json['const']],
-            ));
+        $property->setRequired($this->propertyMetaDataCollection->isAttributeRequired($propertyName));
+
+        $check = $property->isRequired()
+            ? '$value !== ' . var_export($json['const'], true)
+            : '!in_array($value, ' . RenderHelper::varExportArray([$json['const'], null]) . ', true)';
```

Comment:
This basically allows always `null` for optional const values which is strictly not valid. The model generator has the `implicitNull` setting (https://php-json-schema-model-generator.readthedocs.io/en/latest/gettingStarted.html#implicit-null) to behave in a more PHPish-way (basically exactly what you implemented). If this option is not enabled, null should not be an acceptable value for an optional property with a const constraint.